woocommerce/woocommerce
# Add basic UI and functionality for background removal to the Media Library

* Add hook for AI Image Background Removal API requests.### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Adds remove background option to Media Library photos.

Closes #40440  .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Products -> Add new
2. Click featured image to display modal
3. Select image, or upload new one
4. Click "Remove background" link in the top of the sidebar
5. After a few seconds the new image should appear in the Media Library
6. If you select that image, it will _not_ allow you to remove the background again
7. Try to add an image over 10mb, and remove it's background. It should give you an error.


### Screenshots

![image](https://github.com/woocommerce/woocommerce/assets/444632/37749740-a45c-4976-9243-5a2eea19b0ad)
